Sources/TuistCore/ContentHashing/ContentHasher.swift

```diff
@@ -62,7 +62,7 @@ public struct ContentHasher: ContentHashing {
             return try await fileHandler.contentsOfDirectory(filePath)
                 .filter { filesFilter($0) }
                 .sorted(by: { $0 < $1 })
-                .serialMap { try await hash(path: $0) }
+                .concurrentMap(maxConcurrentTasks: 100) { try await hash(path: $0) }
```

Comment:
We need to limit the concurrency here as otherwise, we'd easily exhaust the number of opened handles (see: https://github.com/tuist/tuist/pull/6840). Increasing the number of concurrent tasks wouldn't help much anyway as we have a limited number of threads.

Reply:
I have mixed feelings with limiting at this level. While it's easy to reason about how high the number is with few of these "concurrent" calls, it'll get tricky as we add more of these. Is this something that we can address for example at the `FileSystem` level once we migrate all the calls to it?

Reply:
I think this limiting would make sense regardless of the `FileSystem` issue, you don't want to overload the system with too many tasks asking for threads and the number of files can be potentially really high.

But yes, this can and should be also addressed at the `FileSystem` level